test/unit/io/sequence_file/sequence_file_format_genbank_test.cpp

```diff
@@ -134,239 +147,42 @@ REFERENCE   1  (bases 1 to 5028)
-    {
-R"(LOCOS ID1	stuff
-DEFINITION  Homo sapiens mRNA for prepro cortistatin like peptide, complete
-            cds.
-ACCESSION   ID1
-ORIGIN
-        1 ACGTTTTTTT TTTTTTTT
-//
-)"
-    };
-
-    std::stringstream istream{input};
-    seq.clear();
-    EXPECT_THROW( (format.read(istream, options, seq, std::ignore, std::ignore)), parse_error);
-
-}
-
-TEST_F(read, seq_qual)
-{
-    std::stringstream istream{input};
-    sequence_file_input_options<dna5, true> options2;
-
-    std::vector<qualified<dna5, phred42>> seq_qual;
-
-    for (unsigned i = 0; i < 3; ++i)
-    {
-        id.clear();
-        seq_qual.clear();
-
-        EXPECT_NO_THROW( (format.read(istream, options2, seq_qual, id, seq_qual) ));
-
-        EXPECT_TRUE((ranges::equal(id, expected_ids[i])));
-        EXPECT_TRUE((ranges::equal(seq_qual | views::convert<dna5>, expected_seqs[i])));
-    }
-}
-
-TEST_F(read, illegal_alphabet)
-{
-    std::string input
-    {
-R"(LOCUS ID1	stuff
-DEFINITION  Homo sapiens mRNA for prepro cortistatin like peptide, complete
-            cds.
-ACCESSION   ID1
-ORIGIN
-        1 ACGTTTT?TT TTTTTTTT
-//
-)"
-    };
-
-    std::stringstream istream{input};
-    EXPECT_THROW(( format.read(istream, options, seq, id, std::ignore)), parse_error );
-}
-
-TEST_F(read, from_stream_file)
-{
-    sequence_file_input fin{std::istringstream{input}, format_genbank{}, fields<field::SEQ, field::ID>{}};
-
-    size_t counter = 0;
-    for (auto & [ seq, id ] : fin)
-    {
-        EXPECT_TRUE((std::ranges::equal(seq,  expected_seqs[counter])));
-        EXPECT_TRUE((std::ranges::equal(id,  expected_ids[counter])));
-
-        counter++;
-    }
-
-    EXPECT_EQ(counter, 3u);
+    do_read_test(standard_input);
 }
 
 // ----------------------------------------------------------------------------
 // writing
 // ----------------------------------------------------------------------------
 
-struct write : public ::testing::Test
+struct write : public sequence_file_data
 {
-    std::vector<dna5_vector> seqs
-    {
-        { "ACGTTTTTTTTTTTTTTT"_dna5 },
-        { "ACGTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTT"_dna5 },
-        { "ACGTTTA"_dna5 },
-    };
-
-    std::vector<std::string> ids
-    {
-        "ID1",
-        "ID2",
-        "ID3"
-    };
-
-    std::string comp
-    {
-R"(LOCUS       ID1                 18 bp
-ORIGIN
-        1 ACGTTTTTTT TTTTTTTT
-//
-LOCUS       ID2                 82 bp
-ORIGIN
-        1 ACGTTTTTTT TTTTTTTTTT TTTTTTTTTT TTTTTTTTTT TTTTTTTTTT TTTTTTTTTT
-       61 TTTTTTTTTT TTTTTTTTTT TT
-//
-LOCUS       ID3                 7 bp
-ORIGIN
-        1 ACGTTTA
-//
-)"
-    };
-
-    detail::sequence_file_output_format_REMOVEME<format_genbank> format;
-
     sequence_file_output_options options;
 
     std::ostringstream ostream;
 
     void do_write_test()
     {
+        sequence_file_output fout{ostream, format_genbank{}, fields<field::SEQ, field::ID>{}};
+        fout.options = options;
```

Comment:
also options are not initialized yet. Is that a problem?

Reply:
doesn't seem so but I initialised all of them